traj_opt/examples/example_base.cc

```diff
@@ -201,23 +213,62 @@ TrajectoryOptimizerSolution<double> TrajOptExample::SolveTrajectoryOptimization(
   SolverParameters solver_params;
   SetSolverParameters(options, &solver_params);
 
-  // Establish an initial guess
-  std::vector<VectorXd> q_guess = MakeLinearInterpolation(
-      opt_prob.q_init, options.q_guess, opt_prob.num_steps + 1);
+  std::vector<VectorXd> q_guess;
+  if (options.use_demonstration) {
+    // TODO: Support absolute path
+    const std::string demo_file = "drake/traj_opt/examples/demo.yaml";
+    // Load yaml file with initial guess
+    Demonstration data = yaml::LoadYamlFile<Demonstration>(
+        FindResourceOrThrow(demo_file));
+
+    std::vector<VectorXd> q_guess_tmp = data.observations;
+    // create a vector of input times with the same size as q_guess and delta = 0.01
+    std::vector<double> input_time(q_guess_tmp.size());
+    for (std::size_t i = 0; i < q_guess_tmp.size(); i++) {
+      input_time[i] = i * 0.01;
+    }
+    std::vector<double> opt_time_values(opt_prob.num_steps + 1);
+    for (int i = 0; i < options.num_steps + 1; i++) {
+      opt_time_values[i] = i * options.time_step;
+    }
+
+    std::vector<MatrixXd> q_knots(q_guess_tmp.size(), VectorXd(nq));
+    for (std::size_t i = 0; i < q_guess_tmp.size(); ++i) {
+      q_knots[i] = q_guess_tmp[i];
+    }
+
+    PiecewisePolynomial<double> traj = PiecewisePolynomial<double>::CubicWithContinuousSecondDerivatives(
+                  input_time, q_knots);
+    
+    for (int i = 0; i < options.num_steps + 1; i++) {
+      q_guess.push_back(traj.value(opt_time_values[i]));
+    }
+  }
+  else
+  {
+    // Establish an initial guess
```

Comment:
nit, consider moving this comment to the top of this section and say something more specific here, e.g.:
# Otherwise, use the linear interpolation between q_init and q_guess as the initial guess